test/context/testQueueFactory.spec.js

```diff
@@ -0,0 +1,129 @@
+import testQueueFactory from '../../src/context/testQueueFactory'
+import runTest from '../../src/context/runTest'
+
+jest.mock(
+  '../../src/context/pubSubFactory',
+  () => () => {
+    const listeners = new Set()
+    return {
+      clearListeners: jest.fn(() => listeners.clear()),
+      listen: jest.fn((listener) => {
+        listeners.add(listener)
+        return () => listeners.delete(listener)
+      }),
+      notify: jest.fn((...args) => listeners.forEach(ls => ls(...args))),
+    }
+  }
+)
+
+jest.mock(
+  '../../src/context/runTest',
+  () => jest.fn(() => Promise.resolve({ pass: true }))
+)
+
+function delay(time = 0) {
```

Comment:
Sorry I misunderstood the method you were referring... 
For the `delay` method you are right ;) I've updated the code!